### PR TITLE
Add examples and tests for `epinowcast` methods and uncovered plots

### PR DESCRIPTION
**Pull request by seabbs**
This PR adds basic examples and tests for `summary.epinowcast()`, `plot.epinowcast()`, and `enw_plot_obs()` (which was the only plotting function not fully covered by the new tests for `plot.epinowcast()`). 

It also resolves an issue with the default in `enw_plot_quantiles` which was `TRUE` vs `FALSE` as documented.

### CODE REVIEW

File: R/methods.R [hunk 1]
```diff
@@ -61,6 +75,16 @@ summary.epinowcast <- function(object, type = c(
...
+#' nowcast <- enw_example("nowcast")
+#' latest_obs <- enw_example("obs")
+#'
+#' # Plot nowcast
+#' plot(nowcast, latest_obs = latest_obs, type = "nowcast")
+#'
+#' # Plot posterior predictions by reference date
+#' plot(nowcast, type = "posterior_prediction") +
+#'  ggplot2::facet_wrap(ggplot2::vars(reference_date), scales = "free")
 plot.epinowcast <- function(x, latest_obs = NULL, type = c(
```
Comment (pearsonca) [inline R/methods.R:88]: arguably, `ggplot.epinowcast` is more appropriate here.

Comment (pearsonca) [reply R/methods.R:88]: also, to emphasize previously offered opinion: the plotting stuff should be a separate package.

File: R/plot.R [hunk 2]
```diff
@@ -77,14 +86,20 @@ enw_plot_obs_by_report <- function(obs, log = TRUE, ...) {
...
+#' @seealso [enw_plot_nowcast_quantiles()], [enw_plot_pp_quantiles()]
 #' @family plot
 #' @inheritParams enw_plot_obs
 #' @export
-enw_plot_quantiles <- function(posterior, latest_obs = NULL, log = TRUE, ...) {
+#' @examples
+#' nowcast <- enw_example("nowcast")
+#' nowcast <- summary(nowcast, probs = c(0.05, 0.2, 0.8, 0.95))
+#' enw_plot_quantiles(nowcast, x = reference_date)
+enw_plot_quantiles <- function(posterior, latest_obs = NULL, log = FALSE, ...) {
```
Comment (pearsonca) [inline R/plot.R:102]: feels weird. i get this is MVP, but: quantiles should be discovered, rather than hard coded.

**Review by pearsonca — COMMENTED** [review]
Seems generally fine, but also: this should be its own extension package.

File: R/methods.R (hunk 1 above)
Comment (seabbs) [reply R/methods.R:88]: Yes I agree. This depends on developer time and I think is only a good idea when moving out of the experimental stage. If you go multi-repo to early you can tie yourself into a dev nightmare. 

The plan is to do this with all pre and post processing at some point (just not yet)

Comment (seabbs) [reply R/methods.R:88]: > arguably, ggplot.epinowcast is more appropriate here.

Agree, but its very standard practice to use `ggplot2` inside `plot` calls.

File: R/plot.R (hunk 2 above)
Comment (seabbs) [reply R/plot.R:102]: That is what I did in `EpiNow2` and it was a very big pain to dev + maintain. I am happy to see this here though I just don't want to do the dev (at least right now).

Comment (seabbs) [reply R/plot.R:102]: (happy to see an issue on this of course)

**Comment by seabbs** [comment]
@pearsonca do you have a review decision on this? Going to merge in #208 first, resolve merge issues, and then merge this in once ready.

**Review by pearsonca — APPROVED** [review]

**Comment by pearsonca** [comment]
Comfortable with it. Added issue(+) for splitting off vis package + introspecting Qs.

**Pull request merged**